consensus/safety-rules/src/safety_rules.rs

```diff
@@ -107,24 +110,29 @@ impl SafetyRules {
     ///     1. Validator not in the set
     ///     2. Validator in the set, but no matching key found in storage
     fn reconcile_key(&mut self, epoch_state: &EpochState) -> Result<(), Error> {
-        let signer = self.signer()?;
+        // fake a signer using the same author in case we have not initialized
+        let tmp_signer = {
+            let mut rng = StdRng::from_seed([8u8; 32]);
+            ValidatorSigner::new(self.author, Ed25519PrivateKey::generate(&mut rng))
+        };
```

Comment:
I'm not quite sure if I should do this... This is in case we have not initialized a validator_signer yet, it will generate one with the account address when safety rule is first intialized. Then during intialization, safety rule will reconcile the generated key with the one in persistent storage.. @davidiw

Reply:
I think you're trying too hard to conform this to your original code, instead what you likely want is:

```
let curr_key = self.signer.map(|s| s.public_key()); // this is now either None or Some(public_key);
if curr_key != Some(expected_key) {
   // reconcile
} else {
  // matches
}
```